### PR TITLE
TST: Clean moments consistency

### PR DESCRIPTION
**Pull request by charlesdong1991**
As discussed in https://github.com/pandas-dev/pandas/pull/30577#issuecomment-570014613

This PR is the first step to achieve it, it basically does one thing: break up the giant consistency test in `common` into four tests based on test contents: `var`, `std`, `cov`, and `series data`.

cc @jreback

### CODE REVIEW

**Comment by jreback** [comment]
great looks good. when doing these types of reactoring just make sure that we don't lose tests (a bit tricky to check here as you are actually changing the number of tests), but lgtm.

**Comment by jreback** [comment]
thanks @charlesdong1991

**Comment by charlesdong1991** [comment]
thanks @jreback i will keep followups smaller, and easier to check if any test is lost!

File: pandas/tests/window/moments/test_moments_expanding.py [hunk 1]
```diff
@@ -418,3 +404,44 @@ def test_expanding_consistency(self, min_periods):
                     # GH 9422
                     if name in ["sum", "prod"]:
                         tm.assert_equal(expanding_f_result, expanding_apply_f_result)
+
+    @pytest.mark.parametrize("min_periods", [0, 1, 2, 3, 4])
```
Comment (jbrockmendel) [inline pandas/tests/window/moments/test_moments_expanding.py:408]: should these get `@pytest.mark.slow`?  `test_rolling_consistency_cov` clocks in at 38 seconds for me

**Pull request merged**